SingularityUI/app/application.coffee

```diff
@@ -101,6 +101,20 @@ class Application
             return if @blurred and jqxhr.statusText is 'timeout'
 
             url = settings.url.replace(config.appRoot, '')
+            
+            try
+                serverMessage = JSON.parse(jqxhr.responseText).message or jqxhr.responseText
+            catch
+                serverMessage = jqxhr.responseText
+
+            serverMessage = _.escape serverMessage
+            
+            # Check if the file/directory doesn't exist yet so we
+            # can handle the error with somee style on tailing pages           
+            directoryPatt = new RegExp("does not have a directory yet")
+            hasNoDirectory = directoryPatt.test serverMessage
+            
+            return if jqxhr.status is 400 and hasNoDirectory
```

Comment:
Could you use `app.caughtError()` so that you don't need to duplicate this dirrectoryPattern logic in both places (here and in `LogLines.coffee`?)